pandas/core/sparse/frame.py

```diff
@@ -846,6 +846,13 @@ def applymap(self, func):
         return self.apply(lambda x: lmap(func, x))
 
 
+def _nan_to_np_nan(value):
+    """Normalize nan values to singleton np.NaN object so that when NaNs are
```

Comment:
The existing failing test for sparse is `pandas.tests.sparse.test_frame.TestSparseDataFrame.test_nan_columnname`.
Issue #8822 has some discussion. `nan` in indexes [are supported](https://github.com/pandas-dev/pandas/blob/9c44f9b2cad863bde17c7dd061d5b5b5ccbada21/pandas/core/indexes/base.py#L1828) and have valid uses (e.g. https://github.com/pandas-dev/pandas/issues/3729).
So xfailing that test in this PR ...